RecoParticleFlow/PFClusterProducer/plugins/PFCTRecHitProducer.cc

```diff
@@ -186,20 +183,20 @@ void PFCTRecHitProducer::produce(edm::Event& iEvent,
     //Loop on the calotower constituents and search for HCAL
     double dead = 0.;
     double alive = 0.;
-    for(unsigned int i=0;i< hits.size();++i) {
-      if(hits[i].det()==DetId::Hcal) { 
+    for(auto hit : hits) {
```

Comment:
it seems OK to make a copy here. (```(auto hit : hits)``` operates on copies from the vector)
Was it a manual choice or is there some automated decision based on class content analysis?